packages/coverage-v8/src/provider.ts

```diff
@@ -443,16 +437,11 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
     const filePath = normalize(fileURLToPath(url))
 
     let isExecuted = true
-    let transformResult:
-      | FetchResult
-      | Awaited<ReturnType<typeof this.ctx.vitenode.transformRequest>>
-      = transformResults.get(filePath)
+    let transformResult: FetchResult | TransformResult | undefined = transformResults.get(filePath)
 
     if (!transformResult) {
       isExecuted = false
-      transformResult = await this.ctx.vitenode
-        .transformRequest(filePath)
-        .catch(() => null)
+      transformResult = await onTransform(url.replace('file://', '')).catch(() => undefined)
```

Comment:
I am a bit concerned with transforms like `replace('a', 'b')`. Can we use `fileURLToPath` and `pathToFileURL`?

Reply:
In this specific case we cannot. Here the `url` is a String and may contain query parameters, e.g. in Vue. Using `fileURLToPath` would remove those. If `url` was an instance of `URL` instead of String we could access the `.search` but it's just String here.

```sh
$ pnpm run test --project v8-browser test/vue.test.ts
```

```js
console.log(url)
> 'file:///Users/x/vitest/test/coverage-test/fixtures/src/Vue/Defined.vue?vue&type=style&index=0&scoped=f7f04e08&lang.css',
 
console.log(fileURLToPath(url))
> '/Users/x/vitest/test/coverage-test/fixtures/src/Vue/Defined.vue'

console.log(url.replace('file://', ''))
> '/Users/x/vitest/test/coverage-test/fixtures/src/Vue/Defined.vue?vue&type=style&index=0&scoped=f7f04e08&lang.css'
> ^^This is what we want
```

Reply:
Can we slice it then instead of replacing?

Reply:
Sure! I guess `url.slice(7)` (7 for `file://`.length) would be faster than find-and-replace.